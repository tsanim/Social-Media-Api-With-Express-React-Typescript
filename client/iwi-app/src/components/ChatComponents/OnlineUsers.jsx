import React from 'react';
import PropTypes from 'prop-types';
import UserCard from '../DiscoverComponents/UserCard';

function OnlineUsers(props) {
    const {
        onlineUsers,
    } = props

    return (<div className="onlineUsers">
        <h3>Online users</h3>
        <ul>
            {
                (onlineUsers && onlineUsers.length > 0)
                    ? onlineUsers.map(u => {
                        return (<li key={u._id}>
                            <button id={`${u._id}`} onClick={props.showRoomHandler}>
                                <UserCard
                                    user={u}
                                />
                            </button>
                        </li>)
                    })
                    : null
            }
        </ul>
    </div>
    )
}

OnlineUsers.propTypes = {
    onlineUsers: PropTypes.array,
    showRoomHandler: PropTypes.func
}


export default OnlineUsers;


