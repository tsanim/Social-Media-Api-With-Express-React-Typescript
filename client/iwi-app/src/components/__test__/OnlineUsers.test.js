import React from 'react';
import OnlineUsers from '../ChatComponents/OnlineUsers';

import { shallow } from 'enzyme';

function onlineUsersComponentSetup() {
    const props = {
        showRoomHandler: jest.fn(),
        onlineUsers: [ { _id: '123' } ],
    }

    const enzymeWrapper = shallow(<OnlineUsers {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('Online Users component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = onlineUsersComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div tag container with class "onlineUsers"', () => {
        const { enzymeWrapper } = onlineUsersComponentSetup();

        expect(enzymeWrapper.find('div.onlineUsers')).toHaveLength(1);
    });

    it('should render ul tag', () => {
        const { enzymeWrapper } = onlineUsersComponentSetup();

        expect(enzymeWrapper.find('ul')).toHaveLength(1);
    });

    it('should render h3 tag with text "Online users"', () => {
        const { enzymeWrapper } = onlineUsersComponentSetup();

        expect(enzymeWrapper.find('h3').text()).toEqual('Online users');
    });

    it('should render ul tag with items count equal to onlineUsers array lenght', () => {
        const { enzymeWrapper, props } = onlineUsersComponentSetup();

        expect(enzymeWrapper.find('ul').children()).toHaveLength(props.onlineUsers.length);
    });

    it('should has child ul tag with no items when onlineUsers array is empty', () => {
        const { enzymeWrapper } = onlineUsersComponentSetup();

        enzymeWrapper.setProps({ onlineUsers: [] });

        expect(enzymeWrapper.find('ul').children().getElements()).toHaveLength(0);
    });
});