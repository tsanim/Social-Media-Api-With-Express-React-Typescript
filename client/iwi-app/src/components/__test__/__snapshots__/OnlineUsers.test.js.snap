// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Online Users component should render initial layout 1`] = `
Array [
  <div
    className="onlineUsers"
  >
    <h3>
      Online users
    </h3>
    <ul>
      <li>
        <button
          id="123"
          onClick={[MockFunction]}
        >
          <UserCard
            user={
              Object {
                "_id": "123",
              }
            }
          />
        </button>
      </li>
    </ul>
  </div>,
]
`;
